Tasks/NuGet/Tests/NugetMockHelper.ts

```diff
@@ -0,0 +1,91 @@
+    }
+
+    public registerNugetToolGetterMock() {
+        this.tmr.registerMock('nuget-task-common/NuGetToolGetter', {
+            getNuGet: function(versionSpec) {
+                return "c:\\from\\tool\\installer\\nuget.exe";
+            },
+        });
+    }
+    
+    public registerNugetVersionMock(productVersion: string, versionInfoVersion: number[]) {
+        this.registerNugetVersionMockInternal(productVersion, versionInfoVersion);
+        this.tmr.registerMock('nuget-task-common/pe-parser/index', {
+            getFileVersionInfoAsync: function(nuGetExePath) {
+                let result: VersionInfo = { strings: {} };
+                result.fileVersion = new VersionInfoVersion(versionInfoVersion[0], versionInfoVersion[1], versionInfoVersion[2], versionInfoVersion[3]);
+                result.productVersion = new VersionInfoVersion(versionInfoVersion[0], versionInfoVersion[1], versionInfoVersion[2], versionInfoVersion[3]);
+                result.strings['ProductVersion'] = productVersion;
+                return result;
+            }
+        })
+    }
+
+    private registerNugetVersionMockInternal(productVersion: string, versionInfoVersion: number[]) {
+        this.tmr.registerMock('./pe-parser', {
+            getFileVersionInfoAsync: function(nuGetExePath) {
+                let result: VersionInfo = { strings: {} };
+                result.fileVersion = new VersionInfoVersion(versionInfoVersion[0], versionInfoVersion[1], versionInfoVersion[2], versionInfoVersion[3]);
+                result.strings['ProductVersion'] = productVersion;
+                return result;
+            }
+        })
+    }
+    
+    public registerNugetUtilityMock(projectFile: string[]) {
+        this.tmr.registerMock('nuget-task-common/Utility', {
+            resolveFilterSpec: function(filterSpec, basePath?, allowEmptyMatch?) {
+                return projectFile;
+            },
+            getBundledNuGetLocation: function(version) {
+                return 'c:\\agent\\home\\directory\\externals\\nuget\\nuget.exe';
+            },
+            stripLeadingAndTrailingQuotes: function(path) {
+                return path;
+            },
+            locateCredentialProvider: function(path) {
+                return 'c:\\agent\\home\\directory\\externals\\nuget\\CredentialProvider';
+            },
+            setConsoleCodePage: function() {
+                var tlm = require('vsts-task-lib/mock-task');
+                tlm.debug(`setting console code page`);
+            }
+        } )
+    }
+    
+    public registerNugetConfigMock() {
+        var nchm = require('./NuGetConfigHelper-mock');
+        this.tmr.registerMock('nuget-task-common/NuGetConfigHelper', nchm);
+    }
+    
+    public registerToolRunnerMock() {
+        var mtt = require('vsts-task-lib/mock-toolrunner');
+        this.tmr.registerMock('vsts-task-lib/toolrunner', mtt);
+    }
+    
+    public setAnswers(a) {
```

Comment:
I left this as is.
TaskLibAnswers, the interface on which a is based, is [really used as a generic object](https://github.com/Microsoft/vsts-task-lib/blob/master/node/mock-answer.ts#L30-L57) so the compiler complains when I try to set the osType.